addons/hr_holidays/report/hr_leave_report.py

```diff
@@ -106,3 +107,26 @@ def action_open_record(self):
             'res_id': self.leave_id.id if self.leave_id else self.allocation_id.id,
             'res_model': 'hr.leave' if self.leave_id else 'hr.leave.allocation',
         }
+
+    def _search_has_department_manager_access(self, operator, value):
+        supported_operators = ["!=", "="]
+        if operator not in supported_operators or not isinstance(value, bool):
+            raise NotImplementedError()
+        return [
+                '|',
+                    ('employee_id.user_id', '=', self.env.user.id),
+                    ('employee_id.department_id', 'child_of',
+                        self.env['hr.department']._search([('manager_id', 'in', self.env.user.employee_ids.ids)])
+                    ),
```

Comment:
I don't know if this will be more efficient than to do search :thinking: 

I would like to see sql query for this search :innocent: You can use profiler